### ISSUE TITLE
show_results_immediately not really doing anything

### ISSUE THREAD

**Issue by nullbyto** [anyrun-org/anyrun#200]
As the title states, the option `show_results_immediately` set to true is not actually changing anything. 
When anyrun starts there is only the search bar with no results shown.

**Comment by retrozinndev** [comment]
are you using this option on the main configuration? I mean the `config.ron` file.

**Comment by retrozinndev** [comment]
Just noticed here that this option also stopped working on my system.

**Comment by retrozinndev** [comment]
also, `max_entries` is not being used properly by anyrun. It has the default value of 5.

**Comment by Kirottu** [comment]
The reason `max_entries` in the main Anyrun config does nothing, is because it is not a configuration option for Anyrun itself, it is exclusively used by the plugins.

And as for why `show_results_immediately` doesn't seem to work is because most plugins (except for the Stdin plugin, most prevalently) simply do not provide any results for when there is no input. I basically designed the whole config option just to accomodate the possibility of dmenu-like scripts that would use Anyrun as a selector with only the Stdin plugin enabled.

**Comment by retrozinndev** [comment]
> The reason `max_entries` in the main Anyrun config does nothing, is because it is not a configuration option for Anyrun itself, it is exclusively used by the plugins.

All plugins seem to completely ignore their respective configuration files, that is the problem.

**Comment by Kirottu** [comment]
`show_results immediately` is not a configuration option for the Stdin plugin, it's a configuration option for Anyrun itself.

And as for how you configured `max_entries`, that's incorrect, the Stdin plugin does not use an Option for that value instead it just takes a plain unsigned integer in this fashion:

```ron
Config(
  // Whether to allow the user to input any arbitrary text besides the options provided
  allow_invalid: false,
  // How many entries should be displayed at max
  max_entries: 6,
)
```

**Comment by retrozinndev** [comment]
Works now, thanks!

**Comment by retrozinndev** [comment]
Oh wait, looks like `max_entries` property has 5 as maximum value? Why tho?

**Comment by Kirottu** [comment]
It.. does not? Make sure you actually have enough results to reach your configured number of maximum entries.

**Comment by retrozinndev** [comment]
Yes, I'm sure there are more than 5 entries available, I can search them by typing in the search bar.

**Comment by Kirottu** [comment]
If you clear out the input field do you still not have 6 entries? Also, if the config fails to load Anyrun is gonna output the result if you run it in the terminal (which is the only reason it wouldn't work).

**Comment by retrozinndev** [comment]
> @retrozinndev sorry misread, but I think you have a similar problem- you need to change the max_results both in the plugin config and the base config, otherwise one will limit the other. Seems like there's a chance you haven't updated `config.ron` with a new max amount, maybe?


Will try doing that! Thanks for helping me out

**Comment by retrozinndev** [comment]
> Also, if the config fails to load Anyrun is gonna output the result if you run it in the terminal (which is the only reason it wouldn't work).

Just noticed that the `desktop_actions` field was in fault, placed it in the applications plugin config file and it worked now! Thanks.
